Add recommendation for code formatting (and apply it)

HansOlsson opened the pull request:
Add some description of how we want code to be formatted, and apply the rules.
Some minor items to clean up.

henrikt-ma commented on Modelica/Electrical/Machines/Examples/DCMachines/DCPM_CurrentControlled.mo:
```diff
@@ -90,8 +90,7 @@ equation
   connect(dcpm.pin_an, ground.p)
-    annotation (Line(points={{4,-10},{4,0},{-8,0}},
-                                                   color={0,0,255}));
+    annotation (Line(points={{4,-10},{4,0},{-8,0}},color={0,0,255}));
```
Missing at least this space, by comparison to other annotations seen in this PR diff:
```suggestion
    annotation (Line(points={{4,-10},{4,0},{-8,0}}, color={0,0,255}));
```

HansOlsson commented on Modelica/Electrical/Analog/Examples/Utilities/DirectCapacitor.mo:
```diff
@@ -45,11 +45,9 @@ equation
   connect(currentSource.n, capacitor.p)
-    annotation (Line(points={{-40,10},{0,10}},
-                                                 color={0,0,255}));
+    annotation (Line(points={{-40,10},{0,10}},   color={0,0,255}));
```
```suggestion
    annotation (Line(points={{-40,10},{0,10}}, color={0,0,255}));
```

HansOlsson commented on Modelica/Electrical/Machines/Sensors/CurrentQuasiRMSSensor.mo:
```diff
@@ -39,8 +39,7 @@ equation
   connect(ToPolar1.y[1], Gain1.u) annotation (Line(
-      points={{0,-41},{0,-50.5},{0,-58}},
-                                                          color={0,0,127}));
+      points={{0,-41},{0,-50.5},{0,-58}},                 color={0,0,127}));
```
```suggestion
      points={{0,-41},{0,-50.5},{0,-58}}, color={0,0,127}));
```

HansOlsson commented on Modelica/Electrical/Analog/Examples/Utilities/DirectCapacitor.mo:
```diff
@@ -45,11 +45,9 @@ equation
   connect(ground.p, capacitor.n)
-    annotation (Line(points={{-20,-10},{0,-10}},
-                                                   color={0,0,255}));
+    annotation (Line(points={{-20,-10},{0,-10}},   color={0,0,255}));
```
Should be corrected now.

tobolar commented on Modelica/Fluid/Pipes.mo:
```diff
@@ -940,7 +940,7 @@ This also allows for taking into account friction losses with respect to the act
             // Assumptions
             parameter Boolean allowFlowReversal=system.allowFlowReversal
-		  "= true, if flow reversal is enabled, otherwise restrict flow to design direction (states[1] -> states[n+1])"
+          "= true, if flow reversal is enabled, otherwise restrict flow to design direction (states[1] -> states[n+1])"
```
There are several (many, in fact) cases in this package braking the new rule on indentation, e.g. the file's lines 3 or 901. So either fix them all (tedious) or let us first separate classes into particular files (cf. e.g. https://github.com/modelica/ModelicaStandardLibrary/pull/3433) and fix them afterwards.

HansOlsson replied:
I believe that it should be done later - I just wanted to start by having the rules in place and fix the most obvious issues.

I agree that making all of the non-deliberate(?) indentation consistent will be tedious - I will see if it will be possible to automate (without changing the deliberate) - similarly as it is possible to move classes to separate files; but it will still take time to review.

tobolar replied:
So simply resolve/close this conversation if desired.

tobolar commented on Modelica/Media/IdealGases/Common/package.mo:
```diff
@@ -147,7 +147,7 @@ Temperature T (= " + String(T) + " K) is not in the allowed range
 
       redeclare function extends setSmoothState
-    "Return thermodynamic state so that it smoothly approximates: if x > 0 then state_a else state_b"
+      "Return thermodynamic state so that it smoothly approximates: if x > 0 then state_a else state_b"
```
```suggestion
        "Return thermodynamic state so that it smoothly approximates: if x > 0 then state_a else state_b"
```

tobolar commented on Modelica/Thermal/FluidHeatFlow/Sensors/RelTemperatureSensor.mo:
```diff
@@ -4,7 +4,7 @@ model RelTemperatureSensor "Temperature difference sensor"
   Modelica.Blocks.Interfaces.RealOutput y(unit="K")
     "Temperature difference as output signal"
-	annotation (absoluteValue = false, Placement(transformation(
+ annotation (absoluteValue = false, Placement(transformation(
```
```suggestion
    annotation (absoluteValue = false, Placement(transformation(
```

tobolar commented on ModelicaTest/Electrical/QuasiStatic/Polyphase.mo:
```diff
@@ -94,13 +86,11 @@ package Polyphase "Polyphase quasi-static package"
+    Modelica.Electrical.QuasiStatic.Polyphase.Ideal.Short short(m=m) annotation (Placement(transformation(extent={{-80,20},{-60,40}})));
     Modelica.Electrical.QuasiStatic.Polyphase.Ideal.IdealCommutingSwitch idealCommutingSwitch(m=m,
       Ron=fill(1E-5, m),
       Goff=fill(1E-5, m))                                                                          annotation (Placement(transformation(extent={{30,20},{50,40}})));
```
```suggestion
      Goff=fill(1E-5, m)) annotation (Placement(transformation(extent={{30,20},{50,40}})));
```

tobolar commented on ModelicaTest/Tables/CombiTable2Ds.mo:
```diff
@@ -535,7 +535,7 @@ double mydummyfunc(double dummy_in) {
       table=transpose([1,0,6;2,4,4;3,9,9;4,16,16]),
       smoothness=Modelica.Blocks.Types.Smoothness.ConstantSegments) annotation(Placement(transformation(extent={{-95,-60},{-75,-40}})));
-    equation
+  equation
       connect(combiTable1D.u,clock1.y) annotation(Line(
```
The indentation rule shall be in fact applied also on "connect(..." too. But I see no necessity to fix this in ModelicaTest library.

HansOlsson replied:
Agreed, so wait with that.

tobolar commented:
@HansOlsson I miss a notice how to indentate multi-line declarations.
And, a code example could be helpfull as well.

HansOlsson commented:
> @HansOlsson I miss a notice how to indentate multi-line declarations. And, a code example could be helpfull as well.

I tried to add both of them.

HansOlsson commented:
@casella as new MAP-Lib leader I hope you approve of this.

HansOlsson commented:
@casella @henrikt-ma can you review it - so that it can be merged before release-branch on Friday January 12th?

HansOlsson commented:
I have now rebased it on master and removed three new cases of trailing white-space. It would be good if this could be approved and merged, @casella @henrikt-ma

henrikt-ma commented on Modelica/package.mo:
```diff
@@ -1300,6 +1300,49 @@ be defined like:
+The tab-character shall not be used, since the tab-stops are not standardized.
+</p>
+<p>
+The code in a&nbsp;class shall be indented relative to the class in steps of two spaces;
```
Have we adopted some convention to always follow _a_ by a non-breaking space?  If not, I suggest using a normal space here:
```suggestion
The code in a class shall be indented relative to the class in steps of two spaces;
```

tobolar replied:
I guess this is based on my changes to the documentation in the last years - I use it whenever improving the documentation. 
IMO we have no such rule but this is a common typography rule.

HansOlsson replied:
Seems so, it was also used in #4216 explaining tabs and groups (that you reviewed).

henrikt-ma replied:
Sure it may have slipped by in some other review.  Resolving, as it turned out to intentional and not introduced for the first time in this PR.

henrikt-ma commented on Modelica/package.mo:
```diff
@@ -1300,6 +1300,49 @@ be defined like:
+The code in a&nbsp;class shall be indented relative to the class in steps of two spaces;
+except that the headings <code>public</code>, <code>protected</code>, <code>equation</code>,
+<code>algorithm</code>, and <code>end</code> of class marker shall not be indented.
+The keywords <code>public</code> and <code>protected</code> are headings for a&nbsp;group
```
Similar to above:
```suggestion
The keywords <code>public</code> and <code>protected</code> are headings for a group
```

HansOlsson replied:
Seems so, it was also used in https://github.com/modelica/ModelicaStandardLibrary/pull/4216 explaining tabs and groups (that you reviewed).

henrikt-ma replied:
Sure it may have slipped by in some other review.  Resolving, as it turned out to intentional and not introduced for the first time in this PR.

henrikt-ma reviewed (changes requested):
The conventions look sensible, and I can't spot anything obviously wrong with the included changes.

What I'd like to avoid however, is the risk of styling conventions for the Modelica Standard Library to be taken as conventions that should apply to all Modelica code.  To avoid this, I suggest changing this sentence:
https://github.com/modelica/ModelicaStandardLibrary/blob/0e0f5d3adaa8a7657827c27e2462ab6c9fb1a304/Modelica/package.mo#L2281

Something like this would make more sense to me:
```
<p>A Modelica main package of the Modelica Standard Library should be compliant with the UsersGuide stated in this documentation:</p>
```

By the way, I had never heard of a _Modelica main package_ until I saw this line – is it a generally recognized term?  Does it mean the same as what I'd refer to as an _MSL sub-library_?

henrikt-ma commented:
> I have now rebased it on master and removed three new cases of trailing white-space. It would be good if this could be approved and merged, @casella @henrikt-ma

Please avoid rebasing, as it breaks GitHub conversations linked to commits.  For example, I was unable to find my previous review comments in order to see if and how they had been addressed.

HansOlsson commented:
> > I have now rebased it on master and removed three new cases of trailing white-space. It would be good if this could be approved and merged, @casella @henrikt-ma
> 
> Please avoid rebasing, as it breaks GitHub conversations linked to commits. For example, I was unable to find my previous review comments in order to see if and how they had been addressed.

Rebasing is unfortunately needed to ensure easier merging - as the status was too messy, but that's why it should only be done _after_ addressing the comments.

henrikt-ma reviewed (approved):
OK, the potential risk of these guidelines being misinterpreted as applying to all Modelica code (not just in the MSL) isn't new and will be reported separately.
